packages/replicache/src/transactions.ts

```diff
@@ -271,7 +272,8 @@ export class SubscriptionTransactionWrapper implements ReadTransaction {
  * {@link ReplicacheOptions.mutators} and allows read and write operations on the
  * database.
  */
-export interface WriteTransaction extends ReadTransaction {
+export interface WriteTransaction<TZeroData extends ZeroTxData = ZeroTxData>
```

Comment:
I would prefer if we do not change Replicache public types.

For Reflect we have:

https://github.com/rocicorp/reflect-archive/blob/209cecc2c899bfbe16e57f32dcad1be8f75cf024/packages/reflect-shared/src/types.ts#L21

Reply:
We have the same for Zero:

https://github.com/rocicorp/mono/blob/743c95499e12f0f66119fc819921da85031d4263/packages/zero-client/src/client/custom.ts#L70-L80

https://github.com/rocicorp/mono/blob/743c95499e12f0f66119fc819921da85031d4263/packages/zero-client/src/client/custom.ts#L90-L98

but somehow we need to plumb through the correct IVM fork. Only Replicache has the information required to create that fork (base head hash, new head hash) and when to fork (when rebase begins).